### PR TITLE
Modified etcd manifest to support clustering.

### PR DESCRIPTION
**Pull request by jszczepkowski**
Modified etcd manifest to support clustering.

### CODE REVIEW

**Comment by jszczepkowski** [comment]
This is the fixed version of PR #29284: if variable describing initial etcd cluster is empty (this happens on GKE), we set it to the host name.

**Comment by jszczepkowski** [comment]
@k8s-bot test unit
Issue #29451

**Comment by jszczepkowski** [comment]
@k8s-bot unit test this issue #29451

File: cluster/gce/configure-vm.sh [hunk 1]
```diff
@@ -452,10 +452,11 @@ network_policy_provider: '$(echo "$NETWORK_POLICY_PROVIDER" | sed -e "s/'/''/g")
 enable_manifest_url: '$(echo "${ENABLE_MANIFEST_URL:-}" | sed -e "s/'/''/g")'
 manifest_url: '$(echo "${MANIFEST_URL:-}" | sed -e "s/'/''/g")'
 manifest_url_header: '$(echo "${MANIFEST_URL_HEADER:-}" | sed -e "s/'/''/g")'
-master_name: '$(echo "${MASTER_NAME:-}" | sed -e "s/'/''/g")'
+master_name: '$(echo "${KUBERNETES_MASTER_NAME:-}" | sed -e "s/'/''/g")'
```
Comment (fgrzadkowski) [inline cluster/gce/configure-vm.sh]: Why do you change this line?

File: cluster/gce/configure-vm.sh [hunk 2]
```diff
@@ -452,10 +452,11 @@ network_policy_provider: '$(echo "$NETWORK_POLICY_PROVIDER" | sed -e "s/'/''/g")
...
 num_nodes: $(echo "${NUM_NODES:-}" | sed -e "s/'/''/g")
 e2e_storage_test_environment: '$(echo "$E2E_STORAGE_TEST_ENVIRONMENT" | sed -e "s/'/''/g")'
 kube_uid: '$(echo "${KUBE_UID}" | sed -e "s/'/''/g")'
+initial_etcd_cluster: '$(echo "${KUBERNETES_MASTER_NAME:-}" | sed -e "s/'/''/g")'
```
Comment (fgrzadkowski) [inline cluster/gce/configure-vm.sh]: Should't this use INITIAL_ETCD_CLUSTER var?

Comment (jszczepkowski) [inline cluster/gce/configure-vm.sh]: Fixed.

File: cluster/gce/configure-vm.sh (hunk 1 above)
Comment (jszczepkowski) [inline cluster/gce/configure-vm.sh]: There was an error. Please not that we set KUBERNETES_MASTER_NAME in [common.sh](https://github.com/kubernetes/kubernetes/blob/master/cluster/common.sh#L544). In effect, variable `master_name` had incorrect value in salt config (was empty).

Comment (jszczepkowski) [inline cluster/gce/configure-vm.sh]: Salt variable `master_name` is only used to create name of PV for influxdb, that's why this problem  remained unnoticed for so long.

**Comment by wojtek-t** [comment]
LGTM

**Comment by zmerlynn** [comment]
This PR broke cluster turn-up on AWS. Trying to diagnose and fix it now.

File: cluster/gce/configure-vm.sh (hunk 1 above)
Comment (zmerlynn) [inline cluster/gce/configure-vm.sh]: The use of `master_name` is misguided. The master may not be resolvable even on itself, and this is just not true on AWS:
```
admin@ip-172-20-0-9:~$ ping kubernetes-master
ping: unknown host kubernetes-master
```

I'd like to rip out these pillars completely, and the use in the influxdb PetSet, where it's only used as a uniquifier (instance-prefix is a fine uniquifier).

Comment (zmerlynn) [inline cluster/gce/configure-vm.sh]: And yes, on something like `kops` we might have a hard requirement on DNS, but that's not true on all deployments yet.

**Comment by zmerlynn** [comment]
I put up #30047 as a WIP, but it's not working yet.

**Comment by jszczepkowski** [comment]
Part of https://github.com/kubernetes/features/issues/48

**Pull request merged**